intrastat_product/models/account_invoice.py

```diff
@@ -50,34 +52,19 @@ def _compute_intrastat_country(self):
             inv.src_dest_country_id = country.id
             inv.intrastat_country = country.intrastat
 
-    @api.model
-    def _default_intrastat_transaction_id(self):
-        rco = self.env['res.company']
-        company = rco._company_default_get()
-        inv_type = self._context.get('type')
-        if inv_type == 'out_invoice':
-            return company.intrastat_transaction_out_invoice
-        elif inv_type == 'out_refund':
-            return company.intrastat_transaction_out_refund
-        elif inv_type == 'in_invoice':
-            return company.intrastat_transaction_in_invoice
-        elif inv_type == 'in_refund':
-            return company.intrastat_transaction_in_refund
-        else:
-            return self.env['intrastat.transaction']
-
     @api.model
     def _default_src_dest_region_id(self):
         rco = self.env['res.company']
-        company = rco._company_default_get()
+        company = rco._company_default_get('account.invoice')
```

Comment:
FYI, the proto of _company_default_get shows that the object argument is sort of deprecated: https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_company.py#L167